src/features/recovery/RecoveryManager.ts

```diff
@@ -0,0 +1,409 @@
+                30,
+                session.startTime,
+                false
+            );
+
+            this.#foldingBuffer = new FoldingBuffer();
+
+            const buffer = Buffer.alloc(this.#pageSize);
+            let offset = 0;
+            let recordCount = 0;
+
+            const processChunk = async () => {
+                try {
+                    if (this.#cancelRecovery) {
+                        this.#cancelRecovery = false;
+                        this.#releaseBuffers();
+                        dispatch(setSessionRecoveryPending(false));
+                        onCancel();
+                        return;
+                    }
+
+                    if (offset >= fileSize) {
+                        try {
+                            await dispatch(this.#finalizeRecovery(session));
+                            this.#releaseBuffers();
+                            dispatch(setSessionRecoveryPending(false));
+                            onComplete();
+                        } catch (finalizeError) {
+                            if (finalizeError instanceof Error) {
+                                onFail(finalizeError);
+                            } else {
+                                onFail(
+                                    new Error(
+                                        'Unknown error during finalizeRecovery.'
+                                    )
+                                );
+                            }
+                        }
+                        return;
+                    }
+
+                    const remainingBytes = fileSize - offset;
+                    const bytesToRead = Math.min(
+                        this.#pageSize,
+                        remainingBytes
+                    );
+
+                    this.#fileBuffer
+                        ?.read(buffer, offset, bytesToRead)
+                        .then(() => {
+                            offset += bytesToRead;
+
+                            recordCount = this.#processBuffer(
+                                buffer,
+                                bytesToRead,
+                                recordCount
+                            );
+
+                            const progress = (offset / fileSize) * 100;
+                            onProgress(progress);
+
+                            queueMicrotask(processChunk);
+                        });
+                } catch (error) {
+                    if (error instanceof Error) {
+                        onFail(
+                            new Error(
+                                `Error recovering session: ${error.message}`
+                            )
+                        );
+                        return;
+                    }
+
+                    onFail(
+                        new Error('Error recovering session: Unknown error')
+                    );
+                }
+            };
+
+            queueMicrotask(processChunk);
```

Comment:
Given that processChunk is CPU an IO heavy and that:

"Immediately after every macrotask, the engine executes all tasks from microtask queue, prior to running any other macrotasks or rendering or anything else." -> https://javascript.info/event-loop

I would recommend using a macrotask task (the same as what you did for searchOrphanedSessions) to make UI more responsive.